samza-core/src/main/scala/org/apache/samza/util/CoordinatorStreamUtil.scala

```diff
@@ -110,13 +111,34 @@ object CoordinatorStreamUtil extends Logging {
       jobConfig.getJobId)
   }
 
+  /**
+   * Reads and returns launch config persisted in coordinator stream. Only job.auto sizing configs are currently supported.
+   * @param config full job config
+   * @param metadataStore an instance of the instantiated MetadataStore
+   * @return empty config if auto sizing is disabled, otherwise auto sizing related configs.
+   */
+  def readLaunchConfigFromCoordinatorStream(config: Config, metadataStore: MetadataStore): Config = {
+    if (!config.getBoolean(JobConfig.JOB_AUTOSIZING_ENABLED, false)) {
+      new MapConfig()
+    } else {
+      val config = readConfigFromCoordinatorStream(metadataStore)
+      val launchConfig: util.Map[String, String] = new util.HashMap[String, String]()
+      for ((key:String, value:String) <- config.asScala) {
+        if (key.startsWith(JobConfig.JOB_AUTOSIZING_CONFIG_PREFIX)) {
```

Comment:
Could be simplified along the lines of 

JobConfig jobConfig = new JobConfig(config);
launchConfig = config.map().filter(jobConfig.isAutosizingConfig(key))

Reply:
I was planning to do it, but it seems that this is in Scala and I cannot do

config.entrySet().stream().map().collect()

do you know how to do it in scala?

Reply:
How about
`launchConfig = config.asScala.filterKeys(x => JobConfig.isAutosizingConfig(x))`

Reply:
This is great!